Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -659,7 +659,11 @@ void BlockFormattingContext::layout_block_level_box(Box const& box, BlockContain
         }
     }
 
-    compute_height(box, available_space);
+    // Tables already set their height during the independent formatting context run. When multi-line text cells are involved, using different
+    // available space here than during the independent formatting context run can result in different line breaks and thus a different height.
+    if (!box.display().is_table_inside()) {
+        compute_height(box, available_space);
+    }
```

Comment:
seems like the actual problem you are trying to solve here is that currently for some reason we use wrong available width to calculate table max-content height.

`TableFormattingContext::run()` receives `definite(62.039062) x max-content` as available size from `calculate_max_content_height()`. it is incorrect and should `definite(8) x indefinite` instead (10px - 1px border - 1px border).

maybe you interested in trying to address the root cause of this bug instead of introducing this little hack? :)

Reply:
Are you suggesting to redo `compute_height` with the original used sizes for `box`? Because I had a version of that and it worked, which wasn't surprising. I'm not sure that's what you mean, though - wouldn't that be repeating the exact same computation, with the same used sizes anyway? In which case skipping it is equivalent.

My main question here is why would we call `compute_height` again, I couldn't find hints in the specs that this is indeed the right behavior. I must be overlooking some crucial detail.

Reply:
we should not disable `compute_height` for boxes with TFC but instead we need to make `compute_height` pass correct available width so it computes correct height. does that make sense?

Reply:
That definitely makes sense and it was my initial intuition about why we might want to call `compute_height` again. However, what would be the correct available width for the call to `compute_height` here? And what layout changes would we expect to happen vs. the first layout we did during the width computation (the independent formatting context run)?

Reply:
I added debug logs and it shows that intrinsic layout pass on a table currently uses `definite(62.039062)` which seems wrong consider the table has fixed width equal to 10px, right?

Reply:
If we go with that, which I tend to agree with, why do we need a second call to `compute_height` then? The available width we work with would stay the same, so we'd end up with the same layout anyway. There has to be a reason I'm missing for which the call to `compute_height` isn't redundant.

Another option would be to accept the layout `current_height` currently sets up, but then we have to throw away the cached value in `calculate_min_content_height`, since that is set during the run for determining the width.

Reply:
ok, sorry, I see what you mean here by first and second calls of `compute_height`. currently `compute_height()` is called twice: once from `compute_table_box_width_inside_table_wrapper` during "intrinsic" layout to compute table wrapper width and then another pass to do "normal" layout of table.

though your change doesn't seem to make it call only once but instead disables call `compute_height` for table boxes at all.

Reply:
Yes, that's the gist of it. The problem with calling `compute_height` is that it could decide on a different height for the table box after we've made up our mind about the grid layout. It's unclear to me why we'd want to call it.

The completely right solution might involve two iterations to lift this chicken-and-egg issue between width, height and line breaking and possibly some heuristics. It's not clear to me what's the upside of merely calling `compute_height` here, for tables.